ArchiSteamFarm/Steam/Data/EAssetRarity.cs

```diff
@@ -27,5 +27,11 @@ public enum EAssetRarity : byte {
 	Unknown,
 	Common,
 	Uncommon,
-	Rare
+	Rare,
+	Mythical,
+	Legendary,
+	Ancient,
```

Comment:
`Ancient` is used for both `ancient` and `extraordinary`. I'd prefer latter inside the enumeration for being more general, as `ancient` kind of seems like it would be a reference to some in-game mechanic (e.g. "age" of an item).

Reply:
I'm also leaning more towards `exotic` instead of `legendary`, but the contrast is smaller here anyways. I'm not sure which I'd prefer of `mythical`/`epic`.

Reply:
Also no idea which i'd prefer of `immortal`/`contraband`.

Reply:
What's your opinion, @JustArchi ?

Reply:
> What's your opinion, @JustArchi ?

As enum declaration, we should stick with preferably generic names where rarities aren't tied to game mechanics.

For parsing, we should do the following:

- First of all, match generic names like `Common`, `Uncommon` or `Rare` into existing enums.
- If we're left with any game-specific names, like `Contraband`, we should add another enum value for it, parse to that one, then match with existing one if possible.

I.e.

```csharp
public enum ERarity : byte {
  Common,
  Uncommon,
  Rare,
  Immortal,
  Contraband = Immortal // Or just a number of Immortal if C# won't support this,
  (...)
}
```

Reply:
I'm not expert in rarities and how they match, but I'd say we should use the following logic:

- Put generic names into enum that are sufficient for at least one single game
- When adding new game, start from matching common options
- When left with game-specific options, consider "holes" you didn't match above
- - If you have a natural hole, i.e. one game uses `Immortal`, another `Contraband` in the same place, add `Contraband = Immortal` mapping like above and use it
- - If you have a hole that doesn't match, i.e. game uses rarity that is in-between existing ones, simply add new entry with game-specific name, nobody said every game must use all rarities we have defined.

Reply:
> Or just a number of Immortal if C# won't support this

C# does support this.

Maybe we could also make use of `Enum.TryParse<TEnum>(string? value, bool ignoreCase, out TEnum result)` with `ignoreCase` of `false` 🤔

Reply:
> Could you please elaborate?

If we are already adding all rarieties to the enum (even if just as `Epic = Mythical`), we can make use of standard library behaviour to parse enums from string instead of doing `.ToUpperInvariant()` on the name and having a switch-statement.

Reply:
Ah sorry for the confusion. I mistyped. Of course I meant `ignoreCase` of `true` 😄